skylark/skylark.bzl

```diff
@@ -47,7 +47,7 @@ def _skylark_doc_impl(ctx):
   """Implementation of the skylark_doc rule."""
   skylark_doc_zip = ctx.outputs.skylark_doc_zip
   inputs = _get_transitive_sources(ctx.attr.deps) + ctx.files.srcs
-  sources = [source.path for source in inputs]
+  sources = [source.path for source in ctx.files.srcs]
```

Comment:
This means that I have to explicitly re-mention each source file I want documented in the skydoc rule.

Can you modify skylark_doc() to also accept skylark_library() providers in srcs (or in another property?)

Otherwise, it is impossible to collect skylark_library() rules from different packages into a single skylark_doc() rule; you would also need to specify the source files, which are not individually exported.

Reply:
This stemmed out of an error I was getting building the skylark-docs target, since the cross-repo reference broke the invariant that they shared a common prefix. My assumption is that we want to be explicit about what skylark we want documented, but it makes sense to use a skylark_library as a srcs member and use only the top-level srcs rather than the transitive sources.

If you disagree and we need transitive sources, we'll need to address the common prefix issue a different way.

Reply:
I think omitting transitive sources is fine, although I worry it will break anybody who was using this before...

I also think the path rewriting logic needs to be a bit more complicated to handle libraries, especially split across repos.

Maybe something like this, using `attr.label_keyed_string_dict()`.

```python
skylark_doc(
    name = "my_docs",
    libraries = {
        ":some_library": ".",
        "//other:library": "sub/dir",
    },
)
```

Then, generated documentation for all the source files in `:some_library` winds up in the root of the ZIP file, and `//other:library` winds up in `sub/dir` within the ZIP file.

We'd leave the `srcs` parameter behavior alone entirely, and then eventually remove it with whatever deprecation period is reasonable.

Thoughts?

Reply:
That sounds workable, but is it a use you expect to see? Are these rules supposed to generate docs for the transitive closure, or just the skylark you're introducing in the current directory, such that you need to aggregate other skylark_doc targets in deps if you want to include those docs in addition to your own?

Reply:
Or, more specifically, will every skylark_doc target create its own copy of documentation for skylark_library? :)